[ITEP-70168] Adding Kubevirt and Device-plugin container TAR files to EMT edge-desktop-virtualization image config (for SRIOV VMs)

dmkarthi opened the pull request:
#### Merge Checklist  <!-- REQUIRED -->
**All** boxes should be checked before merging the PR
- [x] The changes in the PR have been built and tested
- [x] cgmanifest file has been updated if required
- [x] Ready to merge

#### Description <!-- REQUIRED -->
Adding Desktop-Virtualization package consisting Kubevirt (v1.5.0 enabled with GTK libarary support and Intel SR-IOV patched QEMU in Virt-Launcher) and Device Plugin(v1) for enabling support of local GTK display using pre-built container tar files. 
This PR provides Kubevirt and Device-plugin tar files as part of DV EMT build image, installation of k3s and its dependencies are not part of this PR.

<!-- Fixes # (issue) -->

#### Any Newly Introduced Dependencies
<!-- Please describe any newly introduced 3rd party dependencies in this change. List their name, license information and how they are used in the project. -->

#### How Has This Been Tested? <!-- REQUIRED -->
Verified installing Kubevirt and device-plugin tar files from extracted location, load them on to k3s containerd environment and apply the opearator and cr files for kubevirt and device-plugin to run on kubernetes cluster, then deployed SR-IOV VM to see it working.

On Immutable and Mutable EMT image, kubevirt and Device-plugin TAR files in desired location

dmkarthi committed:
Add desktop-virtualization for EMT-S

Merge Checklist
All boxes should be checked before merging the PR

 The changes in the PR have been built and tested
 cgmanifest file has been updated if required
 Ready to merge

Description
Add Kubevirt (v1.5.0 enabled with GTK libarary support and Intel SR-IOV patched QEMU in Virt-Launcher) and Device Plugin(v1) for enabling support of local GTK display using pre-built container tar files

Any Newly Introduced Dependencies
How Has This Been Tested?
Installing from tar files from extracted location, load them on to k3s containerd environment and apply the opearator and cr files for kubevirt and device-plugin to run on kubernetes cluster, then deploy VM to see it working

sfonn commented:
please verify functionality is working on immutable image with edge-image-desktop-virtualization.json image.

sfonn commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,35 @@
+Name:           intel-desktop-virtualization-k3s
+Version:        v1.0
```
why is version not same as source tar ball

dmkarthi replied:
made version as 0.1 to match it with tar, will bump to 1.0 once after kubevirt and dp gets final release

sfonn commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,35 @@
+Name:           intel-desktop-virtualization-k3s
+Version:        v1.0
+Release:        1%{?dist}
```
please have url, Vendor and Distribution fields in spec file.
Vendor:         Intel Corporation
Distribution:   Edge Microvisor Toolkit

dmkarthi replied:
Added

dmkarthi commented:
> please verify functionality is working on immutable image with edge-image-desktop-virtualization.json image.

Verified on both Immutable and Mutable EMT image, i can see kubevirt and dp TAR files in desired location, rest of the installation steps from this point onwards is part of cloud init

dmkarthi committed:
versioning and naming update

updated image names and versions

sfonn commented on toolkit/imageconfigs/edge-image-desktop-virtualization-dev.json:
```diff
@@ -77,7 +77,8 @@
                 "packagelists/os-ab-update.json",
-                "packagelists/fs-tools.json"
+                "packagelists/fs-tools.json",
+                "packagelists/intel-desktop-virtualization-k3s.json"
```
Does this feature need to pull in k3s pakages as it is required as part of overall package intel-desktop-virtalization? Not sure if any of below applies for this feature.
https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/134
https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/197

dmkarthi replied:
Scope of the PR is to make kubevirt and Device-plugin tar available as part of DV EMT build image, installation of k3s and its dependencies are not part of this

sfonn commented on toolkit/imageconfigs/edge-image-desktop-virtualization.json:
```diff
@@ -78,7 +78,8 @@
-                "packagelists/fs-tools.json"
+                "packagelists/fs-tools.json",
+                "packagelists/intel-desktop-virtualization-k3s.json"
             ],
```
Does this feature need to pull in k3s pakages as it is required as part of overall package intel-desktop-virtalization? Not sure if any of below applies for this feature.
https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/134
https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/197

dmkarthi replied:
Scope of the PR is to make kubevirt and Device-plugin tar available as part of DV EMT build image, installation of k3s and its dependencies are not part of this

sfonn replied:
But what k3s this is supposed to be working with for EMT for this solution?

anujm1 replied:
> Scope of the PR is to make kubevirt and Device-plugin tar available as part of DV EMT build image, installation of k3s and its dependencies are not part of this

If this expects k3s to be available, those need to be defined as `Requires` in spec

dmkarthi replied:
Added requires k3s

anujm1 commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,36 @@
+URL:            https://github.com/open-edge-platform/edge-desktop-virtualization
+Summary:        Installs Kubevirt (v1.5.0 enabled with GTK libarary support and Intel SR-IOV patched QEMU in Virt-Launcher identified as v1.5.0_DV) and Device Plugin(v1) for enabling support of local GTK display using pre-built container tar files
+
+License:        APACHE 2.0
```
```suggestion
License:        Apache-2.0
```

anujm1 commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,36 @@
+Name:           intel-desktop-virtualization-k3s
+Version:        v0.1
```
```suggestion
Version:        0.1
```

anujm1 commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,36 @@
+
+%files
+/usr/share/%{name}/intel-idv-kubevirt-v0.1.tar.gz
+/usr/share/%{name}/intel-idv-device-plugin-v0.1.tar.gz
```
%{_datadir}

dmkarthi replied:
fixed in all places referred to /usr/share with %{_datadir}

anujm1 commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,36 @@
+BuildArch:      x86_64
+
+%description
+This RPM Installs Kubevirt (v1.5.0 enabled with GTK libarary support and Intel SR-IOV patched QEMU in Virt-Launcher identified as v1.5.0_DV) and Device Plugin(v1) for enabling support of local GTK display using pre-built container tar files
```
Will this always be 1.5.0? I'd suggest rewording and remove "This RPM installs ..." and provide description of the component.

dmkarthi replied:
Removed kubevirt and device-plugin versions in Summary and Description, only kept in change log on v0.1

dmkarthi committed:
Update SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec

Co-authored-by: Anuj Mittal <anuj.mittal@intel.com>

dmkarthi committed:
Update SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec

Co-authored-by: Anuj Mittal <anuj.mittal@intel.com>

dmkarthi committed:
Update intel-desktop-virtualization-k3s.spec

dmkarthi committed:
fixing review comments

dmkarthi committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168

hyunsun commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,36 @@
+%build
+
+%install
+mkdir -p %{buildroot}%{_datadir}/%{name}
```
Save all required container images with `docker save <image>` and the output tar file should be stored under `%{buildroot}%{_sharedstatedir}/rancher/k3s/agent/images`.

Also, k8s manifest to deploy the plugin should be placed under `%{buildroot}%{_sharedstatedir}/rancher/k3s/server/manifests/`.

See https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/230/files for reference.

If it's Helm chart, place the chart tgz file under `%{buildroot}%{_sharedstatedir}/rancher/k3s/server/static/` and create HelmChart CR definition. Please contact me in this case for details.

dmkarthi replied:
The TAR file contains container images and YAML files to enable kubevirt and device-plugin,
[4ff423e](https://github.com/open-edge-platform/edge-microvisor-toolkit/pull/217/commits/4ff423e0dac1edf7b7656aa6424e9d61f940c30a) fixed the extract directory as mentioned

Dhanya122 committed:
Update intel-desktop-virtualization-k3s.spec

krishnajs committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168

cheeyanglee commented on cgmanifest.json:
```diff
@@ -7465,6 +7465,26 @@
+      "component": {
+        "type": "other",
+        "other": {
+          "name": "intel-idv-kubevirt",
```
this seems to me more like 2 different component, shouldn't this be split into two different spec file ?

dmkarthi replied:
both these components are part of single installation, hence made it to single spec file

smitesh-sutaria replied:
Going forward, it does make sense to de-couple them. As device plugins can be used on k3s without the need of kubevirt as well. We will evaluate this for next iteration after customer delivery.

cheeyanglee commented on SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec:
```diff
@@ -0,0 +1,54 @@
+%post
+
+%changelog
+* Fri Jun 20 2025 Dhanya A <dhanya.a@intel.com> - v0.1-2
```
since this is new specs the "release" should be 1 and all the changelog can be merge into single entries for version "0.1-1"

dmkarthi replied:
this versioning is based on https://github.com/open-edge-platform/edge-desktop-virtualization/releases/tag/pre-release-v0.1 and is currently under completion of SDLe process for opensourcing.

dmkarthi committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168

dmkarthi committed:
Update intel-desktop-virtualization-k3s.signatures.json

dmkarthi committed:
Merge branch 'emt-dv-kubevirt-devplugin-ITEP-70168' of https://github.com/dmkarthi/edge-microvisor-toolkit into emt-dv-kubevirt-devplugin-ITEP-70168

cheeyanglee committed:
fix cgmanifest and SPECS

regenerate cgmnifest with
./toolkit/scripts/update_cgmanifest.py first ./cgmanifest.json SPECS/intel-desktop-virtualization-k3s/intel-desktop-virtualization-k3s.spec

Signed-off-by: Lee Chee Yang <chee.yang.lee@intel.com>

dmkarthi committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168

srinidhikasagar commented:
Just a heads up: OSPDT and SDLE are pending approval. ETA is end of June.

cheeyanglee committed:
update LICENSES-MAP.md

Signed-off-by: Lee Chee Yang <chee.yang.lee@intel.com>

cheeyanglee committed:
amend LICENSES-AND-NOTICES/SPECS/data/licenses.json

Signed-off-by: Lee Chee Yang <chee.yang.lee@intel.com>

dmkarthi committed:
changes to spec file to use .tar.zst files

dmkarthi committed:
Merge branch 'emt-dv-kubevirt-devplugin-ITEP-70168' of https://github.com/dmkarthi/edge-microvisor-toolkit into emt-dv-kubevirt-devplugin-ITEP-70168

dmkarthi committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168

sfonn reviewed (approved):
looks good to me

Dhanya122 committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168

anujm1 committed:
Merge branch '3.0-dev' into emt-dv-kubevirt-devplugin-ITEP-70168
